src/field_screen_effect.c

```diff
@@ -73,3 +86,257 @@ void SetFlashScanlineEffectWindowBoundaries(u16 *dest, s32 centerX, s32 centerY,
+#define tFlashRadiusDelta    data[5]
+#define tClearScanlineEffect data[6]
+
+static void UpdateFlashLevelEffect(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+
+    switch (data[0])
+    {
+    case 0:
+        SetFlashScanlineEffectWindowBoundaries(gScanlineEffectRegBuffers[gScanlineEffect.srcBuffer], tFlashCenterX, tFlashCenterY, tCurFlashRadius);
+        data[0] = 1;
+        break;
+    case 1:
+        SetFlashScanlineEffectWindowBoundaries(gScanlineEffectRegBuffers[gScanlineEffect.srcBuffer], tFlashCenterX, tFlashCenterY, tCurFlashRadius);
+        data[0] = 0;
+        tCurFlashRadius += tFlashRadiusDelta;
+        if (tCurFlashRadius > tDestFlashRadius)
+        {
+            if (tClearScanlineEffect == 1)
+            {
+                ScanlineEffect_Stop();
+                data[0] = 2;
+            }
+            else
+            {
+                DestroyTask(taskId);
+            }
+        }
+        break;
+    case 2:
+        ScanlineEffect_Clear();
+        DestroyTask(taskId);
+        break;
+    }
+}
+
+static void sub_807EF7C(u8 taskId)
+{
+    if (!FuncIsActiveTask(UpdateFlashLevelEffect))
+    {
+        EnableBothScriptContexts();
+        DestroyTask(taskId);
+    }
+}
+
+static void sub_807EFA4(void)
+{
+    if (!FuncIsActiveTask(sub_807EF7C))
+        CreateTask(sub_807EF7C, 80);
+}
+
+static u8 sub_807EFC8(s32 centerX, s32 centerY, s32 initialFlashRadius, s32 destFlashRadius, s32 clearScanlineEffect, u8 delta)
+{
+    u8 taskId = CreateTask(UpdateFlashLevelEffect, 80);
+    s16 *data = gTasks[taskId].data;
+
+    tCurFlashRadius = initialFlashRadius;
+    tDestFlashRadius = destFlashRadius;
+    tFlashCenterX = centerX;
+    tFlashCenterY = centerY;
+    tClearScanlineEffect = clearScanlineEffect;
+
+    if (initialFlashRadius < destFlashRadius)
+        tFlashRadiusDelta = delta;
+    else
+        tFlashRadiusDelta = -delta;
+
+    return taskId;
+}
+
+#undef tCurFlashRadius
+#undef tDestFlashRadius
+#undef tFlashRadiusDelta
+#undef tClearScanlineEffect
+
+void AnimateFlash(u8 flashLevel)
+{
+    u8 curFlashLevel = Overworld_GetFlashLevel();
+    u8 value = 0;
+    if (!flashLevel)
+        value = 1;
+    sub_807EFC8(120, 80, sFlashLevelPixelRadii[curFlashLevel], sFlashLevelPixelRadii[flashLevel], value, 2);
+    sub_807EFA4();
+    ScriptContext2_Enable();
+}
+
+void WriteFlashScanlineEffectBuffer(u8 flashLevel)
+{
+    if (flashLevel)
+    {
+        SetFlashScanlineEffectWindowBoundaries(&gScanlineEffectRegBuffers[0][0], 120, 80, sFlashLevelPixelRadii[flashLevel]);
+        CpuFastSet(&gScanlineEffectRegBuffers[0], &gScanlineEffectRegBuffers[1], 480);
+    }
+}
+
+void sub_807F0B0(void)
+{
+    Overworld_FadeOutMapMusic();
+    CreateTask(Task_EnableScriptAfterMusicFade, 80);
+}
+
+static void Task_EnableScriptAfterMusicFade(u8 taskId)
+{
+    if (BGMusicStopped() == TRUE)
+    {
+        DestroyTask(taskId);
+        EnableBothScriptContexts();
+    }
+}
+
+#define tState 9
```

Comment:
>Task_BarnDoorWipeChild has two data objects with different layouts

I don't think that's a real problem. e.g., you can have: 
```
#define tData1 data[1]
#define tData1Alternative data[1]
```
And it won't hurt because it's not redefinition. 
>`DoOutwardBarnDoorWipe` doesn't compare equal if I assign data to a variable and index into it

If there's no `data` local variable, just do things like `gTasks[taskId].tSthYouDefined`